fbs/torchcraft.fbs

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+namespace torchcraft.fbs;
+
+struct Vec2 {
+  x:int;
+  y:int;
+}
+
+table Command {
+  code:int;
+  args:[int];
+  str:string;
+}
+
+table Player {
+  id:int;
+  race:int;              // See torchcraft::BW::Race enum
+  name:string;
+  is_enemy:bool;
+}
+
+table HandshakeClient {
+  protocol:int;           // protocol for binary frame format
```

Comment:
We might want to re-think that comment a bit. Clearly, the switch to FBS warrants a bump in the protocol format but we should formalize what the protocol version actually means if all communication is FBS-based. IIRC we can freely append new fields to the individual tables, for example without breaking backwards compatibility.

Reply:
To re-phrase this a little bit: Previously, the version number was mostly concerned about the frame data blob. Now, the whole communication is FBS tables and we can freely add fields to the end of the tables without worrying about compatibility (as long as we check that respective fields are present before attempting to read them).

Let's assume that we change the structures here sufficiently to warrant a version number update. In this case, would an older version be able to parse them from memory at all? Probably not, as the verification methods might fail (note that I haven't tested this).

Hence, my suggestions for protocols going forward:
- Have an extra round of initial client/server handshake that is limited to version number comparison and maybe some additional client/server info (client ID, for example, or OpenBW vs. SC if we can determine this easily)
- Store a (separate?) protocol version in replayer dumps, in front of the actual data as part of some header.